### PR TITLE
Enable adding keys to and deleting keys from accounts

### PR DESCRIPTION
**Pull request by bowenwang1996**

### CODE REVIEW

File: core/transaction/src/lib.rs [hunk 1]
```diff
@@ -387,6 +462,10 @@ impl From<transaction_proto::SignedTransaction> for SignedTransaction {
                 bytes = t.write_to_bytes();
                 TransactionBody::SwapKey(SwapKeyTransaction::from(t))
             },
+            Some(transaction_proto::SignedTransaction_oneof_body::add_key(t)) => {
+                bytes = t.write_to_bytes();
+                TransactionBody::AddKey(AddKeyTransaction::from(t))
+            }
             _ => unreachable!(),
```
Comment (evgenykuzyakov) [inline core/transaction/src/lib.rs]: No DeleteKey here

File: node/runtime/src/system.rs [hunk 2]
```diff
@@ -189,6 +188,45 @@ pub fn swap_key(
     Ok(vec![])
 }
 
+pub fn add_key(
+    state_update: &mut StateDbUpdate,
+    body: &AddKeyTransaction,
+    account: &mut Account
+) -> Result<Vec<ReceiptTransaction>, String> {
+    let new_key = PublicKey::new(&body.new_key)?;
+    account.public_keys.push(new_key);
```
Comment (evgenykuzyakov) [inline node/runtime/src/system.rs]: We need to check the number of public_keys and prevent adding more than the limit.
Also we need to check that the public_key doesn't exist yet

File: core/transaction/src/lib.rs (hunk 1 above)
Comment (bowenwang1996) [reply core/transaction/src/lib.rs]: Fixed

File: node/runtime/src/system.rs (hunk 2 above)
Comment (bowenwang1996) [reply node/runtime/src/system.rs]: Fixed

**Pull request merged**